patches/minecraft/net/minecraft/world/entity/SpawnPlacements.java.patch

```diff
@@ -26,5 +34,18 @@
 +          if (predicate == null) return net.minecraft.world.level.NaturalSpawner.canSpawnAtBody(this, world, pos, type);
 +          return predicate.test(world, pos, type);
 +      }
++   }
++
++   // ******* FORGE START. INTERNAL USE ONLY! ****** //
++   public static void fireSpawnPlacementEvent()
++   {
++       Map<EntityType<?>, net.minecraftforge.event.entity.SpawnPlacementRegisterEvent.MergedSpawnPredicate> map = Maps.newHashMap();
++       f_21750_.forEach((type, data) -> map.put(type, new net.minecraftforge.event.entity.SpawnPlacementRegisterEvent.MergedSpawnPredicate(data.f_21769_, data.f_21768_, data.f_21767_)));
++       net.minecraftforge.common.ForgeHooks.onSpawnPlacementRegister(map);
++   }
++
++   public static void registerAll(Map<EntityType<?>, net.minecraftforge.event.entity.SpawnPlacementRegisterEvent.MergedSpawnPredicate> map)
++   {
++       map.forEach(((entityType, merged) -> f_21750_.put(entityType, new SpawnPlacements.Data(merged.getHeightmapType(), merged.getSpawnType(), merged.build()))));
```

Comment:
I still don't think you need this second method. You could just add the map loop as the last statement in the `fireSpawnPlacementEvent` since the map instance is the same throughout the calls.